.github/workflows/haskell.yml

```diff
@@ -0,0 +1,96 @@
+name: Haskell CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    defaults:
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ["8.6.5"]
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Select build directory
+      run: echo "CABAL_BUILDDIR=dist" >> $GITHUB_ENV
+
+    - name: Install pkgconfiglite
+      if: matrix.os == 'windows-latest'
+      run: choco install -y pkgconfiglite
+
+    - name: Install libsodium
+      if: matrix.os == 'windows-latest'
+      run: |
+        curl -Ls https://download.libsodium.org/libsodium/releases/libsodium-1.0.18-mingw.tar.gz -o libsodium-1.0.18-mingw.tar.gz
+        tar zxvf libsodium-1.0.18-mingw.tar.gz
+
+        sed -i "s|/d/a/1/s/|D:/a/cardano-node/cardano-node/|g" libsodium-win64/lib/pkgconfig/libsodium.pc
+
+        export PKG_CONFIG_PATH="$(readlink -f libsodium-win64/lib/pkgconfig)"
+        echo "PKG_CONFIG_PATH=$PKG_CONFIG_PATH" >> $GITHUB_ENV
+
+        export LIBSODIUM_PATH="$(readlink -f libsodium-win64/bin | sed 's|^/d|D:|g' | tr / '\\')"
+        echo "LIBSODIUM_PATH=$LIBSODIUM_PATH"
+        echo "$LIBSODIUM_PATH" >> $GITHUB_PATH
+
+    - name: Install libsodium
+      if: matrix.os == 'macos-latest'
+      run: brew install libsodium
+
+    - name: Configure to use libsodium
+      run: |
+        cat >> cabal.project <<EOF
```

Comment:
That is wrong. It should be `cabal.project.local` .

Reply:
The reason I append to `cabal.project` is because `cabal configure` clobbers `cabal.project.local`.

Reply:
If it move it after the `cabal configure` it should work though.

Reply:
Also, `cabal configure` should not be needed with `cabal >= 3.0`. I have not run it in over a year.

Reply:
I hope to eventually introduce `cabal-cache` to speed up build times which needs the `configure` step, so it's useful to keep it separate.

I've noticed the rudimentary cache we use is broken for MacOS and it isn't very good because you need at least one good build (per change in cabal file version) for something to be cached.

Reply:
If you need the separate configure step it would still be better to append the libsodium stuff to `cabal.project.local` (after it is created).

Reply:
Done!